model/src/cfd.rs

```diff
@@ -1438,8 +1438,15 @@ impl Cfd {
                 self.during_contract_setup = false;
             }
             OracleAttestedPostCetTimelock { cet, .. } => self.cet = Some(cet),
-            OracleAttestedPriorCetTimelock { timelocked_cet, .. } => {
+            OracleAttestedPriorCetTimelock {
+                timelocked_cet,
+                commit_tx,
+                ..
+            } => {
                 self.cet = Some(timelocked_cet);
+                if self.commit_tx.is_none() {
+                    self.commit_tx = commit_tx;
```

Comment:
I'm confused: why do we need to set it to `None`? 
Don't you have to check if `commit_tx.is_some()`, otherwise you might set it to `None`?

Reply:
Oh, it does not matter, because it was `none` before :D

Reply:
Yeah, this shows that we should model these events different. The problem is that we put an optional field into an event, but only *some* combinations are valid. Namely, if it was `None` (because we did not have a `ManualCommit`) it *has* to be `Some`. But we cannot enforce that - we cannot just fail `apply` if we don't have a `ManualCommit` and don't have no `commit_tx` in `OracleAttestedPriorCetTimelock`.

I think the correct way to model this is two events, where one contains the `commit_tx` and the other one does not. Depending on the situation we append either the one event or the other. But we cannot easily change events like this.